planner/core/logical_plan_builder.go

```diff
@@ -2312,6 +2313,9 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 	if tblName.L == "" {
 		tblName = tn.Name
 	}
+	if b.getSelectOffset() != -1 {
+		b.ctx.GetSessionVars().PlannerSelectBlockAsName[b.getSelectOffset()] = tblName
```

Comment:
The same question, why we call it `PlannerSelectBlockAsName`.

Reply:
These are dead codes. can be removed.